src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfFeature.java

```diff
@@ -0,0 +1,1320 @@
+     * @return The {@link GencodeGtfFeature} representing the information in {@code gtfLine}
+     */
+    public static GencodeGtfFeature create(String gtfLine) {
+        return create(gtfLine.split("\t"));
+    }
+
+    /**
+     * Create a {@link GencodeGtfFeature} based on a line from a Gencode GTF File.
+     * @param gtfFields A line from a Gencode GTF File split on the {@code \t} character.
+     * @return The {@link GencodeGtfFeature} representing the information in {@code gtfLine}
+     */
+    public static GencodeGtfFeature create(String[] gtfFields) {
+
+        // Ensure that the input data are superficially well-formed:
+        if ( gtfFields.length != GencodeGtfCodec.NUM_COLUMNS ) {
+            throw new UserException.MalformedFile("Invalid number of fields in the given GENCODE line " +
+                    " - Given: " + gtfFields.length + " Expected: " + GencodeGtfCodec.NUM_COLUMNS);
+        }
+
+        GencodeGtfFeature feature = null;
+
+        String featureType = gtfFields[2].toLowerCase();
+
+        // Figure out which kind of feature to make:
+        if ( featureType.equals("gene") ) {
+            feature = GencodeGtfGeneFeature.create(gtfFields);
+        }
+        else if ( featureType.equals("transcript") ) {
+            feature = GencodeGtfTranscriptFeature.create(gtfFields);
+        }
+        else if ( featureType.equals("exon") ) {
+            feature = GencodeGtfExonFeature.create(gtfFields);
+        }
+        else if ( featureType.equals("cds") ) {
+            feature = GencodeGtfCDSFeature.create(gtfFields);
+        }
+        else if ( featureType.equals("utr") ) {
+            feature = GencodeGtfUTRFeature.create(gtfFields);
+        }
+        else if ( featureType.equals("start_codon") ) {
+            feature = GencodeGtfStartCodonFeature.create(gtfFields);
+        }
+        else if ( featureType.equals("stop_codon") ) {
+            feature = GencodeGtfStopCodonFeature.create(gtfFields);
+        }
+        else if ( featureType.equals("selenocysteine") ) {
+            feature = GencodeGtfSelenocysteineFeature.create(gtfFields);
+        }
+        else {
+            throw new UserException.MalformedFile("Unknown type of GencodeGtfFeature: " + featureType);
+        }
+
+        return feature;
+    }
+
+    // ================================================================================================
+
+    @Override
+    public String getContig() {
+        return chromosomeName;
+    }
+
+    @Override
+    public int getStart() {
+        return genomicStartLocation;
+    }
+
+    @Override
+    public int getEnd() {
+        return genomicEndLocation;
+    }
+
+    // ================================================================================================
+
+    @Override
+    public boolean equals(Object other) {
+
+        boolean isEqual = other instanceof GencodeGtfFeature;
```

Comment:
I don't want the subtypes to be equivalent if they're not the same class.  I read the article and it makes some good points, however in this case because some of the fields can have sub-fields, it doesn't make sense to me to have them equal eachother.

That said, I need to refactor the equals methods based on the 'struct-like' class refactoring anyway.